Sming/Arch/Esp8266/Components/driver/uart.cpp

```diff
@@ -360,31 +365,31 @@ static void IRAM_ATTR handle_uart_interrupt(uint8_t uart_nr, uart_t* uart)
 			 * The interrupt gets re-enabled by a call to uart_read() or uart_flush()
 			 */
 			if(bitRead(usis, UIOF)) {
-				bitClear(USIE(uart_nr), UIOF);
+				CLEAR_PERI_REG_MASK(UART_INT_ENA(uart_nr), UART_RXFIFO_OVF_INT_ENA);
 			} else if(read == 0) {
-				USIE(uart_nr) &= ~(_BV(UIFF) | _BV(UITO));
+				CLEAR_PERI_REG_MASK(UART_INT_ENA(uart_nr), UART_RXFIFO_FULL_INT_ENA | UART_RXFIFO_TOUT_INT_ENA);
 			}
 		}
 
 		// Unless we replenish TX FIFO, disable after handling interrupt
-		if(bitRead(usis, UIFE)) {
+		if(usis & UART_TXFIFO_EMPTY_INT_ST) {
```

Comment:
Why not continue to use bitRead here?

Reply:
The definitions for `UART_TXFIFO_EMPTY_INT_ST`, etc. are all bit masks rather than bit numbers so have to do it this way. i.e. 
UART_TXFIFO_EMPTY_INT_ST = 0x02
UIFE = 1